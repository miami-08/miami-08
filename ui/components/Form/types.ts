import { FormikConfig, FormikValues } from 'formik';
import { RouteComponentProps } from 'react-router-dom';
import TNullable from 'types/TNullable';

export interface IFormInputProps {
    label: string;
    name: string;
    type?: string;
}

export interface IInnerFormProps extends TWithoutSubmit {
    title: string;
    fields: IFormInputProps[];
    handleSubmit: (
        values: FormikValues,
        history: RouteComponentProps['history'],
    ) => void;
    errorText?: TNullable<string>;
}

type TWithoutSubmit = Omit<FormikConfig<any>, 'onSubmit'>;
