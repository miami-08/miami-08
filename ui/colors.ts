export const colors = {
    primary: '#39A2DB',
    lightPrimary: '#43BDFF',
    success: '#34A69A',
    warning: '#f0ad4e',
    danger: '#DF5E5E',
    white: '#FFFFFF',
    black: '#1E1E1E',
    gray: '#999999',
    lightGray: '#eaeaea',
};
