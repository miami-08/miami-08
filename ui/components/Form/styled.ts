import styled from 'styled-components';

export const DynamicFormBox = styled.div`
    box-shadow: 0 0 56px rgba(151, 151, 151, 0.38);
    border-radius: 34px;
    max-width: 340px;
    margin: 30px auto auto;
    padding: 50px 30px 20px;

    & button {
        width: 100%;
    }
`;

export const FieldsWrapper = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    align-items: center;
`;

export const Title = styled.h1`
    font-size: 32px;
    line-height: 32px;
    margin: 0;
`;
