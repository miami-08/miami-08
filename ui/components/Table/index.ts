export * from './Table';

export * from './TableElements/TableElements';
