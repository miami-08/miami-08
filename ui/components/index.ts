export * from './BaseButton';
export * from './Input';
export * from './BackButton';
