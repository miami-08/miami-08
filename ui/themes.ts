import { colors } from "./colors";
import { buttonViews } from "./components/Button/buttonViews";
import { inputViews } from "./components/Input/inputViews";

export interface ITheme {
    theme: IThemeType;
}

interface IThemeType {
    colors: IElement;
}

interface IElement {
    body: string;
    text: string;
    buttons: any;
    inputs: any;
}

export const themes = {
    light: {
        colors: {
            body: colors.gray,
            text: colors.white,
            buttons: buttonViews.light,
            inputs: inputViews.light,
        },
    },
    sea: {
        colors: {
            body: colors.lightPrimary,
            text: colors.primary,
            buttons: buttonViews.sea,
            inputs: inputViews.sea,
        },
    },
};
