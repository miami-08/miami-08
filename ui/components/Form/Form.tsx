import React from 'react';
import { Formik, Form } from 'formik';

import { BaseButton, Input, StyledError } from 'ui/components';

import { IInnerFormProps } from 'uicomponents/Form/types';

import * as Styled from './styled';

export const InnerForm: React.FC<IInnerFormProps> = ({
    fields,
    title,
    initialValues,
    validationSchema,
    handleSubmit,
    errorText,
}) => (
    <Formik
        onSubmit={handleSubmit}
        initialValues={initialValues}
        validationSchema={validationSchema}
    >
        {({
            values, errors, touched, handleChange,
        }) => (
            <Form>
                <Styled.DynamicFormBox>
                    <Styled.Title>{title}</Styled.Title>
                    <Styled.FieldsWrapper>
                        {fields!.map((el) => (
                            <Input
                                key={el.name}
                                label={el.label}
                                name={el.name}
                                value={values[el.name]}
                                type={el.type || 'text'}
                                onChange={handleChange}
                                errorText={
                                    errors[el.name] && touched[el.name]
                                        ? errors[el.name]
                                        : null
                                }
                            />
                        ))}
                    </Styled.FieldsWrapper>
                    <BaseButton type="submit">Присоединиться</BaseButton>
                    <StyledError>{errorText}</StyledError>
                </Styled.DynamicFormBox>
            </Form>
        )}
    </Formik>
);
