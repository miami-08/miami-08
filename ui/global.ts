import { createGlobalStyle } from 'styled-components';
import { ITheme } from './themes';

export const GlobalStyles = createGlobalStyle<ITheme>`
html, #root, .app, body {
  height: 100%;
}

body {
    padding: 0;
    color: ${({ theme }) => theme.colors.text};
    background: ${({ theme }) => theme.colors.body};
    margin: 0;
    font-family: 'Roboto', sans-serif;
    
}
`;
