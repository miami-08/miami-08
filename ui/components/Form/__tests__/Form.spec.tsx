import 'jsdom-global/register';
import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { themes } from 'ui/themes';
import { UIForm as Form } from 'ui/components/Form';

const { light } = themes;

afterEach(() => {
    cleanup();
});

describe('Form', () => {
    it('should render initial values in inputs ', async () => {
        const initialValues = {
            login: 'admin',
            password: 'somepassword',
        };

        const fields = [
            {
                label: 'Логин',
                name: 'login',
            },
            {
                label: 'Пароль',
                name: 'password',
                type: 'password',
            },
        ];

        const mockHandle = jest.fn();

        render(
            <ThemeProvider theme={light}>
                <Form
                    handleSubmit={mockHandle}
                    fields={fields}
                    initialValues={initialValues}
                    title="Some title"
                />
            </ThemeProvider>,
        );

        const loginFieldValue: HTMLInputElement = screen
            // @ts-ignore
            // говорит, что value нет у HTMLElement
            .getByLabelText(fields[0].label).value;

        expect(loginFieldValue).toEqual(initialValues.login);
    });
});
