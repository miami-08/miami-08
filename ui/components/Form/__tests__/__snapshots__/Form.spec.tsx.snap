// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Is equal to snapshot 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="sc-pGacB hfruWC"
  >
    <h1
      className="sc-kEjbQP bXatqG"
    >
      Title
    </h1>
    <div
      className="sc-jrAFXE XiYva"
    >
      <div
        className="sc-gsTEea cKwXyU"
      >
        <input
          className="sc-dlfnuX hnVzCV"
          id="form-input-login"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="form-input-login"
        >
          Логин
        </label>
        <div
          className="sc-hKgJUU kYByEP"
        />
      </div>
      <div
        className="sc-gsTEea cKwXyU"
      >
        <input
          className="sc-dlfnuX hnVzCV"
          id="form-input-password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <label
          htmlFor="form-input-password"
        >
          Пароль
        </label>
        <div
          className="sc-hKgJUU kYByEP"
        />
      </div>
    </div>
    <button
      className="sc-bdfBQB eRsREP"
      type="submit"
    >
      Присоединиться
    </button>
    <div
      className="sc-hKgJUU kYByEP"
    />
  </div>
</form>
`;
